tools/awps-tunnel/client/src/panels/RequestHistory.tsx

```diff
@@ -126,28 +128,65 @@ export function RequestHistory(props: RequestHistoryProps) {
   );
 }
 
+const renderContent = (rawString: string, index: number): ReactNode => {
+  try {
+    const parsedJson = JSON.parse(rawString.substring(index));
```

Comment:
check the content-type first, json content-type would start with `application/json` or `text/json`, maybe as one output parameter of parsing the raw request

Reply:
solved with new commit

Reply:
Can you clarify how should I treat them differently in detail? Since I think both `application/json` and `text.json` have the similar format and could be parse into json objects

Reply:
resolved in new commit.